dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -369,6 +369,52 @@ def diff(self, other_profile: NumericStatsMixin, options: Dict = None) -> Dict:
                 "and '{}'".format(cls.__name__, other_profile.__class__.__name__)
             )
 
+        self_bin_counts = self._stored_histogram["histogram"]["bin_counts"]
+        regen_histogram = False
+
+        if isinstance(self_bin_counts, np.ndarray):
+            regen_histogram = True
+
+        new_self_histogram = {"bin_counts": None, "bin_edges": None}
+        new_other_histogram = {"bin_counts": None, "bin_edges": None}
+
+        if regen_histogram:
+            len_self_bin_counts = 0
+            if self_bin_counts.all() is not None and len(self_bin_counts) > 0:
+                len_self_bin_counts = len(self_bin_counts)
+
+            # re-calculate `self` histogram
+            if not len_self_bin_counts == 10 and self_bin_counts.all() is not None:
+                histogram, hist_loss = self._regenerate_histogram(
+                    bin_counts=self._stored_histogram["histogram"]["bin_counts"],
+                    bin_edges=self._stored_histogram["histogram"]["bin_edges"],
+                    suggested_bin_count=10,
+                )
+                new_self_histogram["bin_counts"] = histogram["bin_counts"]
+                new_self_histogram["bin_edges"] = histogram["bin_edges"]
+
```

Comment:
I think there's are only set if _stored_histogram doesn't have len `10`

Reply:
working this

Reply:
good call out here -- now configured to assign already existing `_stored_histogram` to the `new_self_histogram` and/or `new_other_histogram` depending on the conditions that exist.

Reply:
This should almost never occur because of 
```
self.max_histogram_bin: int = 100000
self.min_histogram_bin: int = 1000
```

used in `self._get_histogram()`

Reply:
once that is opened up to the user, for sure